src/main/java/net/logstash/logback/appender/AbstractLogstashTcpSocketAppender.java

```diff
@@ -506,23 +598,80 @@ public synchronized void start() {
             errorCount++;
             addError("No encoder was configured for appender " + name + ".");
         }
-        if (port <= 0) {
-            errorCount++;
-            addError("No port was configured for appender " + name + ".");
+        
+        if( hostAddresses != null && remoteHost != null ) {
+        	errorCount++;
+        	addError("Use 'remoteHost' or 'remoteAddresses' but not both");
         }
-
-        if (remoteHost == null) {
-            errorCount++;
-            addError("No remote host was configured for appender " + name + ".");
-        }
-
-        if (errorCount == 0) {
-            try {
-                InetAddress.getByName(remoteHost);
-            } catch (UnknownHostException ex) {
-                addError("unknown host: " + remoteHost);
-                errorCount++;
-            }
+        else {
+	        if( hostAddresses != null ) {
+	        	String[] hostAndPorts = StringUtils.split(hostAddresses, ",");
+	        	
+	        	hostInfo = new HostInfo[hostAndPorts.length];
+	        	int hostIndex = 0;
+	        	
+	        	for(String hostAndPort: hostAndPorts) {
+	        		String[] parts = StringUtils.split(hostAndPort,':');
+	        		
+	        		if( parts.length != 2 ) {
+	        			errorCount++;
+	        			addError("Invalid hostAddress '"+hostAndPort+"': unparseable value (expected format 'host:port')");
+	        			continue;
+	            	}
+	        		
+	        		String host = StringUtils.trim(parts[0]);
+	                try {
+	                    InetAddress.getByName(host);
+	                } catch (UnknownHostException ex) {
+	                    addError("Invalid hostAddress '"+hostAndPort+"': host unkown (was '"+host+"')");
+	                    errorCount++;
+	                    continue;
+	                }
+	                
+	                int port = 0;
+	                try {
+	                	port = Integer.parseInt(StringUtils.trim(parts[1]));
+	                }
+	                catch(NumberFormatException e) {
+	                	 addError("Invalid hostAddress '"+hostAndPort+"': unparseable port (was '"+parts[1]+"')");
+	                     errorCount++;
+	                     continue;
+	                }
+	        		if( port <= 0 ) {
+						addError("Invalid hostAddress '"+hostAndPort+"': port must be greater than 0 (was "+port+")");
+						errorCount++;
+						continue;
+	                }
+	                
+	                hostInfo[hostIndex] = new HostInfo(host, port);
+	                hostIndex++;
+	        	}
+	        	
+	        	if( port > 0 || remoteHost != null ) {
```

Comment:
The default port is > 0, so this will always trigger a warning.

Just check if `remoteHost != null` here.